### PR TITLE
Create robinInterfaceHelper class

### PR DESCRIPTION
**Pull request by philipcardiff**
The idea here is to create a dedicated robinInterfaceHelper to encapsulate all the Robin coupling changes required in the pimpleFluid solver (to limit code repetition).

@iagolessa what do you think about this idea? This should make it easier to port the Robin approach to the other versions of pimpleFluid.

As a by the way, most of the commits here should not be part of this pull request; I will remove them.

### CODE REVIEW

File: src/solids4FoamModels/fluidModels/pimpleFluid/pimpleFluid.foamextend.C [hunk 1]
```diff
@@ -138,6 +138,13 @@ pimpleFluid::pimpleFluid
 
     setRefCell(p(), pimple().dict(), pRefCell_, pRefValue_);
     mesh().schemesDict().setFluxRequired(p().name());
+
+    if (useFoamExtend40formulation_ && robin_.patchIDs().size() > 0)
+    {
+        FatalErrorIn(type())
+            << "Currently, the Robin coupling approach only works with "
+            << "useFoamExtend40formulation turned off" << abort(FatalError);
```
Comment (iagolessa) [inline src/solids4FoamModels/fluidModels/pimpleFluid/pimpleFluid.foamextend.C]: From this message, I am not sure whether the Robin BCs do not work only with the formulation of FE40 or when it is compiled with FE40. I remember using the Robin conditions with FE40 too, so I believe they work fine with FE40, actually, unless you did any other modification that I am not aware of. Also, in the pEqn file for FE40 the modifications to account for those BCs are included and should work.

File: src/solids4FoamModels/fluidModels/pimpleFluid/pimpleFluid.esi.C [hunk 2]
```diff
@@ -192,6 +193,9 @@ bool pimpleFluid::evolve()
                 {
                     #include "meshCourantNo.H"
                 }
+
+                // Update flux in FSI interface with Robin BC
+                robin_.setInterfaceFluxToZero(phi);
```
Comment (iagolessa) [inline src/solids4FoamModels/fluidModels/pimpleFluid/pimpleFluid.esi.C:198]: In this portion, you started to modify the ESI but is not complete yet right? I mean, the other functions of robinInterfaceHelper are still missing from the code.

**Comment by philipcardiff** [comment]
> @philipcardiff I reviewed the code and the new robinInterfaceHelper.C is an excelent idea to use with pimpleFluid for other OF versions. I added just some minor comments with doubts about some portions of the code.
> I started to analyze the ESI and Foundation versions of pimpleFluid that used to work with the Robin BCs from prior versions of the code and there are some extra modifications to the code apart from the functions that are used in FE41 and included in the robinInterfaceHelper class. I will try to add these changes in the next few days and test whether they work, of course, already using the interface class.

Great, thanks @iagolessa, sorry I have not replied about this; I have a last minute proposal deadline :smile:

**Comment by philipcardiff** [comment]
I will close this as it has gone stale and the robin code has been updated. This can be re-opened if needed.

**Pull request closed without merging**